src/Xamarin.Android.Build.Tasks/Tasks/AppendCustomMetadataToItemGroup.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks {
+	public class AppendCustomMetadataToItemGroup : Task {
+		[Required]
+		public ITaskItem[] Inputs { get; set; }
+
+		[Required]
+		public ITaskItem[] MetaDataItems { get; set; }
+
+		[Output]
+		public ITaskItem[] Output { get; set; }
+
+		public override bool Execute ()
+		{
+			var output = new List<ITaskItem> ();
+
+			foreach (var item in Inputs) {
+				var fn = Path.GetFileNameWithoutExtension (item.ItemSpec);
+				output.Add (item);
+				foreach (var metaData in MetaDataItems) {
+					if (metaData.ItemSpec != fn)
```

Comment:
We should compare in a case-insensitive manner (`OrdinalIgnoreCase`).  This matches Windows and (default) macOS filesystem semantics.  This could plausibly trip up Linux if case-sensitive support were required, but such support *couldn't* work elsewhere, and it complicates things, so...  Just go with case-insensitive.